lib/thrift/generator.ex

```diff
@@ -93,6 +94,48 @@ defmodule Thrift.Generator do
     end)
   end
 
+  defp resolve_name_collisions(generated_modules) do
+    Enum.reduce(generated_modules, [], fn({name, quoted}, acc) ->
+      Keyword.update(
+        acc,
+        name,
+        quoted,
+        fn(existing) -> resolve_name_collision(name, existing, quoted) end
+      )
+    end)
+  end
+
+  defp resolve_name_collision(name, q1, q2) do
+    {meta1, ast1} = get_meta_and_ast(q1)
+    {meta2, ast2} = get_meta_and_ast(q2)
+
+    context1 = Keyword.get(meta1, :context)
+    context2 = Keyword.get(meta2, :context)
+    combined_ast = [name, [do: {:__block__, [], ast1 ++ ast2}]]
+
+    # the context will be the generating module
+    # we can combine constants into other modules, but no other combinations
+    # furthermore, we want to keep whichever context is _not_ the constant
+    # generator so that subsequent combinations will work properly
+    cond do
```

Comment:
I'll definitely update the comments to hopefully make it clearer.

I was just thinking about the TEnum case and I'm not sure it will be easy to work that out cleanly... It should work MOST of the time, but suppose someone has

```
// user_status.thrift
enum UserStatus {
  ACTIVE,
  BANNED
}

const i32 One = 1
```
It's a little convoluted, but we would end up with both a `def one` and a `defmacro one` in the same `UserStatus` module.  Resolving _that_ would be a lot more involved.  We could either just disallow merging TEnum and Constant or allow it with the caveat that this edge case could come up (why would someone define an enum and a const with the same name?).